kclvm/compiler_base/error/src/diagnostic/components.rs

```diff
@@ -54,3 +57,350 @@ impl Component<DiagnosticStyle> for Label {
+    fn format(&self, sb: &mut StyledBuffer<DiagnosticStyle>, _: &mut Vec<ComponentFormatError>) {
+        sb.appendl(&self.content, self.style);
+    }
+}
+
+/// `IndentWithPrefix` is a component of diagnostic to display an indent with prefix.
+/// An indent is a whitespace.
+/// ```ignore
+/// "|   " is three indent with prefix "|".
+/// ```
+pub struct IndentWithPrefix {
+    indent: usize,
+    prefix: StringWithStyle,
+}
+
+const DEFAULT_INDENT_PREFIX_LABEL: &str = "|";
+
+impl IndentWithPrefix {
+    /// You can new a `IndentWithPrefix` by default label with 0 indent.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use compiler_base_error::Component;
+    /// # use compiler_base_error::components::IndentWithPrefix;
+    /// # use compiler_base_error::DiagnosticStyle;
+    /// # use rustc_errors::styled_buffer::StyledBuffer;
+    ///
+    /// let mut sb = StyledBuffer::<DiagnosticStyle>::new();
+    /// let mut errs = vec![];
+    ///
+    /// // If you want to render default text: "|"
+    /// let indent = IndentWithPrefix::default();
+    /// indent.format(&mut sb, &mut errs);
+    /// ```
+    pub fn default() -> Self {
+        Self {
+            indent: 0,
+            prefix: StringWithStyle::new_with_no_style(DEFAULT_INDENT_PREFIX_LABEL.to_string()),
+        }
+    }
+
+    /// You can new a `IndentWithPrefix` by default label with custom indents.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use compiler_base_error::Component;
+    /// # use compiler_base_error::components::IndentWithPrefix;
+    /// # use compiler_base_error::DiagnosticStyle;
+    /// # use rustc_errors::styled_buffer::StyledBuffer;
+    ///
+    /// let mut sb = StyledBuffer::<DiagnosticStyle>::new();
+    /// let mut errs = vec![];
+    ///
+    /// // If you want to add 3 indents and render text: "   |"
+    /// let indent = IndentWithPrefix::new_with_default_label(3, None);
+    /// indent.format(&mut sb, &mut errs);
+    /// ```
+    pub fn new_with_default_label(indent: usize, style: Option<DiagnosticStyle>) -> Self {
+        Self {
+            indent,
+            prefix: StringWithStyle::new_with_style(DEFAULT_INDENT_PREFIX_LABEL.to_string(), style),
+        }
+    }
+
+    /// You can new a `IndentWithPrefix` by custom label with custom indents.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use compiler_base_error::Component;
+    /// # use compiler_base_error::components::IndentWithPrefix;
+    /// # use compiler_base_error::DiagnosticStyle;
+    /// # use rustc_errors::styled_buffer::StyledBuffer;
+    ///
+    /// let mut sb = StyledBuffer::<DiagnosticStyle>::new();
+    /// let mut errs = vec![];
+    /// // If you want to add 3 indents and rendering text: "   ^"
+    /// let indent = IndentWithPrefix::new("^".to_string(), 3, None);
+    /// indent.format(&mut sb, &mut errs);
+    /// ```
+    pub fn new(prefix: String, indent: usize, prefix_style: Option<DiagnosticStyle>) -> Self {
```

Comment:
`prefix` or `prefix_label` or `label` ? Compare to line 263

Reply:
Fixed